test/specs/spaniel-observer.spec.js

```diff
@@ -232,4 +238,35 @@ describe('SpanielObserver', function() {
       })
       .then(cleanUp);
   });
+
+  it('duration should be correct when starting from a backgrounded tab, opening the tab, then closing the tab', function() {
+    spaniel.__w__.document = { visibilityState: 'hidden' };
+
+    return runTest(
+      {
+        label: 'impression',
+        ratio: 0.5,
+        time: 1
+      },
+      {
+        BACKGROUND_TAB_FIX: true
+      }
+    )
+      .then(function(result) {
+        result.observer.onTabShown();
```

Comment:
I think there needs to first be a period of time (like 100ms) before we show the tab. We want to test that the timer didn't start to early too.